rust/lance/src/dataset.rs

```diff
@@ -5768,6 +5768,189 @@ mod tests {
         assert_eq!(results.num_rows(), 1);
     }
 
+    #[tokio::test]
+    async fn test_fts_unindexed_data_on_empty_index() {
+        let test_uri = TempStrDir::default();
+
+        // Empty dataset with fts index
+        let params = InvertedIndexParams::default();
+        let title_col = GenericStringArray::<i32>::from(Vec::<&str>::new());
+        let content_col = GenericStringArray::<i32>::from(Vec::<&str>::new());
```

Comment:
This could just be `StringArray`: 
```suggestion
        let title_col = StringArray::from(Vec::<&str>::new());
        let content_col = StringArray::from(Vec::<&str>::new());
```